### PR TITLE
Clean up ChatColor & add translateAlternateColorCodes();

### PR DESCRIPTION
**Pull request by IsaiahPatton**

### CODE REVIEW

**Comment by CodeCrafter47** [comment]
Your `translateAlternateColorCodes` implementation isn't the most efficient.

If you want to use replaceAll you can do it with a single call:
```java
translatedText = textToTraslate.replaceAll("(?i)" + alt + "([0-9a-fklmnor])", "\u00A7\\1");
```

This works because replaceAll uses the regular expressions.

Alternatively you could implement it similar to the `StripColor` method which would be the most efficient, but that's up to you.

File: PluginReference/src/main/java/PluginReference/ChatColor.java [hunk 1]
```diff
@@ -5,7 +5,8 @@
  */ 			
 public class ChatColor {
 	public static String ColorPrefix   = "\u00A7";
-	public static String BLACK	   = ColorPrefix + "0";
+       public static char   ColorChar     = '\u00A7';
+	public static String BLACK	       = ColorPrefix + "0";
```
Comment (CodeCrafter47) [inline PluginReference/src/main/java/PluginReference/ChatColor.java]: There's a tab in that line which breaks the alignment.

File: PluginReference/src/main/java/PluginReference/ChatColor.java [hunk 2]
```diff
@@ -51,33 +53,16 @@ public static String StripColor(String str) {
...
-                                .replaceAll(altColorChar+"k", MAGIC);
-                                .replaceAll(altColorChar+"l", BOLD);
-                                .replaceAll(altColorChar+"m", STRIKETHROUGH);
-                                .replaceAll(altColorChar+"n", UNDERLINE);
-                                .replaceAll(altColorChar+"o", ITALIC);
-                                .replaceAll(altColorChar+"r", RESET);
-		return translatedText;
-	}
+    
+    public static String translateAlternateColorCodes(String altChar, String textToTranslate) {
```
Comment (CodeCrafter47) [inline PluginReference/src/main/java/PluginReference/ChatColor.java]: Why use String for altChar, not char?

File: PluginReference/src/main/java/PluginReference/ChatColor.java [hunk 3]
```diff
@@ -3,51 +3,47 @@
...
-	public static String AQUA = ColorPrefix + "b";
-	public static String RED = ColorPrefix + "c";
-	public static String LIGHT_PURPLE = ColorPrefix + "d";
-	public static String YELLOW = ColorPrefix + "e";
-	public static String WHITE = ColorPrefix + "f";
-	public static String MAGIC = ColorPrefix + "k";
-	public static String BOLD = ColorPrefix + "l";
+public class ChatColor {
+	public static String ColorPrefix   = "\u00A7";
+        public static char   ColorChar     = '\u00A7';
```
Comment (CodeCrafter47) [inline PluginReference/src/main/java/PluginReference/ChatColor.java]: I think there's one space to much here

File: PluginReference/src/main/java/PluginReference/ChatColor.java [hunk 4]
```diff
@@ -3,51 +3,47 @@
...
+	public static String BLACK	   = ColorPrefix + "0";
```
Comment (CodeCrafter47) [inline PluginReference/src/main/java/PluginReference/ChatColor.java]: Please use spaces instead of tabs.

**Comment by IsaiahPatton** [comment]
YAY! merged!

**Pull request merged**